sceptre/template_handlers/s3.py

```diff
@@ -0,0 +1,43 @@
+import logging
+
+from sceptre.template_handlers import TemplateHandler
+
+
+class S3(TemplateHandler):
+    """
+    Template handler that can resolve templates from S3.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.logger = logging.getLogger(__name__)
+        super(S3, self).__init__(*args, **kwargs)
+
+    def schema(self):
+        return {
+            "type": "object",
+            "properties": {
+                "path": {"type": "string"}
+            },
+            "required": ["path"]
+        }
+
+    def handle(self):
```

Comment:
I left it out for now as the scope is already pretty large. Also, it becomes hairy to support .j2 and .py if the template is not written to disk somewhere. The Jinja and Python support of the `file` handler assume the file is available on disk, so we cannot use that directly.

It would be ugly to have to implement the logic in every handler, I agree. But I have the feeling that refactoring the whole .j2 / .py stuff is out of scope for this PR. Adding that is a step that can be done later, and from what I have seen, it might be really complex to implement.

Reply:
@ngfgrant Wondering if you had any suggestions on how to more easily support .j2 and .py in a DRY manner?